crates/sui-pg-db/src/db.rs

```diff
@@ -0,0 +1,547 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::ops::{Deref, DerefMut};
+use std::time::Duration;
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use chrono::NaiveDateTime;
+use diesel::migration::{MigrationSource, MigrationVersion};
+use diesel::pg::Pg;
+use diesel::prelude::*;
+use diesel::sql_types::BigInt;
+use diesel::ExpressionMethods;
+use diesel::OptionalExtension;
+use diesel_async::async_connection_wrapper::AsyncConnectionWrapper;
+use diesel_async::{
+    pooled_connection::{
+        bb8::{Pool, PooledConnection},
+        AsyncDieselConnectionManager,
+    },
+    AsyncConnection, AsyncPgConnection, RunQueryDsl,
+};
+use scoped_futures::ScopedBoxFuture;
+use sui_indexer_alt_framework_store_traits::{
```

Comment:
nit (optional): It may be cleaner to just import this like so:

```
use sui_indexer_alt_framework_store_traits as store;
```

And then refer to everything qualified by `store::`.